library/src/main/java/com/schibsted/spain/barista/BaristaRecyclerViewActions.java

```diff
@@ -24,21 +26,21 @@ public static void clickRecyclerViewItem(@IdRes int recyclerViewId, int... posit
   }
 
   private static void performClick(@IdRes int recyclerViewId, int position) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
         RecyclerViewActions.actionOnItemAtPosition(position, clickUsingPerformClick()));
   }
 
   public static void scrollTo(int recyclerViewId, int position) {
-    onView(withId(recyclerViewId)).perform(scrollToPosition(position));
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(scrollToPosition(position));
   }
 
   public static void clickRecyclerViewItemChild(@IdRes int recyclerViewId, int position, @IdRes int itemToClickId) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
         RecyclerViewActions.actionOnItemAtPosition(position, clickChildWithId(itemToClickId)));
   }
 
   public static void clickRecyclerViewItemChild(@IdRes int recyclerViewId, int position, String text) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
```

Comment:
@despinola Let me try to improve my English... hehehe @Sloy talked about creating a method that returns the `allOf(...)` Matcher, and talked about applying to Assertions.

Right now, we're only applying it to Actions, to fix the issue you described at #48 .

And then, focusing on applying the new Matcher on the Assertions, I think that it's not needed by now and, in the future, it could be an issue in the future. So I suggest to avoid applying the new suggested method to Assertions, cos it could produce worse tests.